### PR TITLE
Don't zip `governanceProcedures` with infinite list in Tally

### PR DESCRIPTION
**Pull request by teodanciu**
# Description

The test is hanging in the version before this change.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Tally.hs [hunk 1]
```diff
@@ -184,7 +184,7 @@ tallyTransition = do
                   st'
         updateState st'' xs
 
-  updateState st $ Seq.zip govProcedures [0 ..]
+  updateState st $ Seq.zip govProcedures (fmap GovernanceActionIx [0 .. fromIntegral (Seq.length govProcedures - 1)])
```
Comment (lehins) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Tally.hs]: `GovernanceActionIx` has a `Num` instance, so no extra conversions is needed.
```suggestion
  updateState st $ Seq.zip govProcedures [0 .. fromIntegral (Seq.length govProcedures - 1)]
```

**Review by lehins — APPROVED** [review]
Yeah, good catch, using infinite lists for Seq is not a good idea :smile:

File: eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Tally.hs (hunk 1 above)
Comment (teodanciu) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Tally.hs]: Would it be maybe better to do 
`Seq.zip govProcedures (Seq.fromList [0 .. fromIntegral (Seq.length govProcedures - 1)])` 
and remove the OverloadedLists ?

Comment (lehins) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Tally.hs]: They are one and the same thing. This is exactly what `OverloadedLists` will translate it into. So it is up to you.

**Pull request merged**